### PR TITLE
Using a JSON parser with a source map

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: cli.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import * as yargs from "yargs"
 import { log } from "./lib/util/logging"
 
 const defaultLogDir = log.directory
-const logFilepath = log.filepath
+// const logFilepath = log.filepath
```
Comment (vladbarosan) [inline cli.ts]: not used  ? then can we remove

File: lib/validators/modelValidator.ts [hunk 2]
```diff
@@ -27,9 +26,9 @@ const HttpRequest = msRest.WebResource
 
 export class ModelValidator extends SpecValidator<SpecValidationResult> {
 
-  private sampleRequest: Unknown = {}
+  // private sampleRequest: unknown = {}
 
-  private sampleResponse: Unknown = {}
+  // private sampleResponse: unknown = {}
```
Comment (vladbarosan) [inline lib/validators/modelValidator.ts]: remove ?

File: lib/validators/specResolver.ts [hunk 3]
```diff
@@ -385,21 +384,21 @@ export class SpecResolver {
...
-      )
+      // TODO: doc should have a type
+      utils.setObject(doc as {}, slicedRefName, node)
+      const slicedLocalReferenceValue = parsedReference.localReference.value.slice(1)
       let referencedObj = self.visitedEntities[slicedLocalReferenceValue]
       if (!referencedObj) {
         // We get the definition/parameter from the relative file and then add it (make it local)
         // to the doc (i.e. self.specInJson) being processed.
-        referencedObj = utils.getObject(result, slicedLocalReferenceValue)
+        referencedObj = utils.getObject(result, slicedLocalReferenceValue) as SchemaObject
```
Comment (vladbarosan) [inline lib/validators/specResolver.ts]: >getObject [](start = 30, length = 9)

can't we type it ?

**Comment by vladbarosan** [comment]
>   "version": "0.4.67",

68

---
Refers to: package-lock.json:3 in 59961d5. [](commit_id = 59961d5494a3ba475cae89e19526b59eef21df2a, deletion_comment = False)

File: lib/util/utils.ts [hunk 4]
```diff
@@ -640,9 +632,9 @@ interface Entity {
   type?: DataType
   additionalProperties?: SchemaObject | boolean
   items?: SchemaObject
-  "x-nullable"?: any
+  "x-nullable"?: boolean
```
Comment (vladbarosan) [inline lib/util/utils.ts]: >boolean [](start = 17, length = 7)

can this be ever a string ? I saw lower it its bool or string

File: lib/util/utils.ts [hunk 5]
```diff
@@ -108,25 +108,31 @@ export function parseContent(
   filePath: string,
   fileContent: string
 ): SwaggerObject {
-  let result = null
   const sanitizedContent = stripBOM(fileContent)
   if (/.*\.json$/gi.test(filePath)) {
-    result = JSON.parse(sanitizedContent)
+    //  return JSON.parse(sanitizedContent)
```
Comment (vladbarosan) [inline lib/util/utils.ts]: >//  return JSON.parse(sanitizedContent) [](start = 4, length = 39)

remove ?

File: lib/util/utils.ts (hunk 4 above)
Comment (sergey-shandar) [reply lib/util/utils.ts]: it can but we convert it to Boolean. We need a good Schema validator :-)

**Pull request merged**